eppo_core/src/events.rs

```diff
@@ -1,79 +1,96 @@
-use std::collections::HashMap;
+use std::{collections::HashMap, sync::Arc};
 
-use serde::{Deserialize, Serialize};
+use serde::Serialize;
 
-use crate::{eval::eval_details::EvaluationDetails, Attributes};
+use crate::{eval::eval_details::EvaluationDetails, Attributes, SdkMetadata, Str};
 
 /// Events that can be emitted during evaluation of assignment or bandit. They need to be logged to
 /// analytics storage and fed back to Eppo for analysis.
-#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[derive(Debug, Clone, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Events {
     pub assignment: Option<AssignmentEvent>,
     pub bandit: Option<BanditEvent>,
 }
 
-/// Represents an event capturing the assignment of a feature flag to a subject and its logging
-/// details.
-#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+/// Common fields for the same split.
+#[derive(Debug, Clone, Serialize)]
 #[serde(rename_all = "camelCase")]
-pub struct AssignmentEvent {
+pub struct AssignmentEventBase {
     /// The key of the feature flag being assigned.
-    pub feature_flag: String,
+    pub feature_flag: Str,
     /// The key of the allocation that the subject was assigned to.
-    pub allocation: String,
+    pub allocation: Str,
     /// The key of the experiment associated with the assignment.
     pub experiment: String,
```

Comment:
Should `experiment` be an `Str` too?

Reply:
It doesn't need to be because it's never cloned. `String` is slightly better because `format!()` returns `String`

Reply:
Why is experiment never cloned, but allocation is?

Reply:
Because we clone `allocation` _from_ parsed UFC config but `experiment` is created for each split. After `AssignmentEventBase` is constructed, the whole event base is reused (via Arc)—we don't clone individual fields